docs/design/milestone_2.md

```diff
@@ -0,0 +1,260 @@
+# Criticality Score Revamp: Milestone 2 - Productionizing
+
+- Author: [Caleb Brown](mailto:calebbrown@google.com)
+- Updated: 2022-11-18
+
+## Goals
+
+The primary goal of Milestone 2 is to ensure the fresh criticality score data is
+always available and enable continuous improvement on how the scores are
+calculated.
+
+To achieve this the criticality score system will be productionized to
+continuously update the criticality score for GitHub repositories with at least
+20 stars.
+
+### Non-Goals
+
+The same non-goals for
+[Milestone 1](https://github.com/ossf/criticality_score/blob/main/docs/design/milestone_1.md)
+apply to this milestone:
+
+- Improving the score
+- Covering source repositories on non-GitHub hosts
+- De-duping mirrors
+
+While these are important, they are beyond the scope of this milestone.
+
+## Background
+
+This work is a follow up to
+[Milestone 1](https://github.com/ossf/criticality_score/blob/main/docs/design/milestone_1.md).
+
+The Criticality Score project is used by the OpenSSF and the Securing Critical
+Projects Work Group (WG) for determining which Open Source projects are
+"critical". Critical Open Source projects are those which are broadly depended
+on by organizations, and present a security risk to those organizations, and
+their customers, if they are not supported.
+
+Improving the criticality score requires introducing new data sources and
+signals and collecting them across all repositories.
+
+Additionally the open source ecosystem is dynamic, and an unassuming project can
+quickly become critical if it becomes a dependency of one or two popular
+projects.
+
+### Key Challenges
+
+There are two key challenges with collecting and scoring critical projects:
+
+1. Duration
+2. API token management
+
+Duration makes it impractical for an individual to run the signal data
+themselves across a large number of open source projects. An interruption or
+failure across multiple days or weeks needed to complete the signal collection
+requires constant monitoring.
+
+Efficiently collecting signals requires access to many GitHub API tokens and
+monitoring of quotas to avoid unnecessary delays when quotas are met. Multiple
+tokens also allows greater parallelism and faster collection.
+
+Building centralized infrastructure for automating the collection of signals
+across all open source projects helps address these challenges.
+
+### Scorecard Infrastructure
+
+The Scorecard project has already built out automation for collecting and
+aggregating data across hundreds of thousands of repositories, as documented in
+[Scalable Scorecard](https://github.com/ossf/scorecard/blob/main/docs/design/scalable_scorecards.md).
+
+Productionizing the collection of signals for calculating a criticality score
+aligns well with this infrastructure.
+
+## Design
+
+### Enumeration
+
+![GitHub Enumeration Infra Design](images/github-enumeration-infra.svg)
```

Comment:
Maybe link to the code for the entry point ?

Reply:
Done